src/test/java/test/retryAnalyzer/RetryAnalyzerTest.java

```diff
@@ -90,6 +89,17 @@ public void testIfRetryIsInvokedWhenTestMethodHas(Class<?> clazz, int size, Map<
         };
     }
 
+    @Test(description = "GITHUB-1538")
+    public void testIfDependentMethodsAreInvokedWhenRetrySucceeds() {
+        TestNG testng = create(TestClassSampleWithTestMethodDependencies.class);
+        InvokedMethodNameListener listener = new InvokedMethodNameListener();
+        testng.addListener(listener);
+        testng.run();
+        assertThat(listener.getSucceedMethodNames()).containsExactly("a", "b");
+        assertThat(listener.getFailedMethodNames()).containsExactly("a");
+        assertThat(listener.getSkippedMethodNames()).containsExactly("a");
```

Comment:
I wonder if the skipped methods should not be: `"b", "a", "b"`

Reply:
I didn't quite get you there @juherr. Since there's a retry analyzer involved , the method `a()` would have an entry only in passed and skipped list (it having an entry in failed list is due to how our `InvokedMethodNameListener` is built. I got rid of using it and instead fell back to using `TestListenerAdapter`. So skipped would never have `b` in it, because `b()` passes in the very first attempt.

Reply:
My remark is maybe out of the scope of this pull request, but I'm not understanding the [maybe current] behavior:

I don't understand what is the method called flow. Could you detail it?

Reply:
Current behavior : 

Method `b()` has a dependency on `a()`
Method `a()` has a retry analyzer associated with it.

`a()` initially fails, and then passes eventually due to `retry analyzer`
`b()` is expected to run, but does not run, because its only considering the `skipped`/`failed` status of `a()`  and not the eventual `pass` status.

I fixed this, by having TestNG check if `a()` also had a `pass` and if yes, allow it to run `b()`

Please let me know if that adds clarity @juherr

Reply:
But what is the flow?
Is it: `a` (failed/skip), `a` (pass), `b` (pass)? 
=> the `a` from failed or skipped method should be removed.

IMO, a more natural flow could be:
`a` (failed/skip), `b` (skip), `a` (pass), `b` (pass).